dspace-api/src/main/java/org/dspace/ctask/general/BasicLinkChecker.java

```diff
@@ -132,10 +132,18 @@ protected int getResponseStatus(String url) {
         try {
             URL theURL = new URL(url);
             HttpURLConnection connection = (HttpURLConnection) theURL.openConnection();
-            int code = connection.getResponseCode();
-            connection.disconnect();
+            connection.setInstanceFollowRedirects(true);
+            int statusCode = connection.getResponseCode();
+            if ((statusCode == HttpURLConnection.HTTP_MOVED_TEMP || statusCode == HttpURLConnection.HTTP_MOVED_PERM ||
+                    statusCode == HttpURLConnection.HTTP_SEE_OTHER)) {
+                connection.disconnect();
+                String newUrl = connection.getHeaderField("Location");
+                if (newUrl != null) {
+                    return getResponseStatus(newUrl);
```

Comment:
(after group review)

A badly configured site (infinite redirects) could mean this recursive function never completes. We could add a new configuration property (could just be a curation task parameter that you enforce https://wiki.lyrasis.org/display/DSDOC7x/Curation+System, or a DSpace configuration property?) that gives a max redirects integer, with 0 or -1 meaning no limit. (my suggestion would be 0 for "do not follow any redirects" and -1 for "no limit", just for the most flexibility)